src/NUnitFramework/framework/Constraints/Tolerance.cs

```diff
@@ -169,10 +169,14 @@ public Tolerance Ticks
         public object Amount { get; }
 
         /// <summary>
-        /// Returns true if the current tolerance has not been set or is using the .
+        /// Returns true if the current tolerance has not been set or is using the default.
         /// </summary>
         public bool IsUnsetOrDefault => Mode == ToleranceMode.Unset;
 
+        /// <summary>
+        /// Returns true if the current tolerance varies from exact and default. Indicating tolerance needs processing.
+        /// </summary>
+        public bool HasVariance => Mode != ToleranceMode.Unset && this != Exact && this != Default;
```

Comment:
It's public because my default was to use what's used for properties of the same usage/type.  So I copied IsUnsetOrDefault.  Also other Tolerance implementers might want to be able to find out if the Tolerance HasVariance or not.  Especially useful for comparers.  If it's internal, how would a custom AreEqual constraint check if there is a variance?

In my many different attempts to write NodaTime extensions for NUnit, I have had to replace the tolerance with a custom implementation, as well as the EqualConstraint to make it work.  Extending Equals has been much more difficult that I would have hoped.  It's difficult to add an automatic addition for a new type.  Restricting access to this value makes maintenance easier, but I believe limits those who wish to extend NUnit.

I'm not sure how much extending of NUnit goes on, but my first attempt ran into a number of things that made it harder than I thought it would be.

Reply:
Thanks, that's a good point. I hadn't thought about custom comparers but they're certainly relevant even within the scope of this issue. That makes sense to me, thanks!

As a general rule, I'd suggest "internal by default" because public members are much more difficult to change in a non-breaking way, but I can see why this should be public. 👍